ssl/quic/quic_port.c

```diff
@@ -1329,13 +1347,78 @@ static int port_validate_token(QUIC_PKT_HDR *hdr, QUIC_PORT *port,
         *scid = hdr->src_conn_id;
     }
 
+    /*
+     * Determine if we need to send a NEW_TOKEN frame
+     * If we validated a retry token, we should always
+     * send a NEW_TOKEN frame to the client
+     *
+     * If however, we validated a NEW_TOKEN, which may be
+     * reused multiple times, only send a NEW_TOKEN frame
+     * if the existing received token has 10% of its lifetime
+     * remaining.  This prevents us from constantly sending
+     * NEW_TOKEN frames on every connection when not needed
+     */
+    if (token.is_retry) {
+        *gen_new_token = 1;
+    } else {
+        if (time_diff < NEW_TOKEN_LIFETIME / 10)
```

Comment:
If I understand it correctly, this should instead be something like:
```suggestion
        if (time_diff > NEW_TOKEN_LIFETIME - (NEW_TOKEN_LIFETIME / 10))
```
So that time_diff will be greater than the top 90% of NEW_TOKEN_LIFETIME.

Also small nit: this can be an `else if` statement